contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol

```diff
@@ -223,28 +223,26 @@ contract FunctionsRouter is RouterBase, IFunctionsRouter, FunctionsSubscriptions
    * @inheritdoc IFunctionsRouter
    */
   function fulfill(
-    bytes32 requestId,
     bytes memory response,
     bytes memory err,
     uint96 juelsPerGas,
     uint96 costWithoutFulfillment,
-    address transmitter
+    address transmitter,
+    IFunctionsRequest.Commitment memory commitment
   ) external override returns (uint8 resultCode, uint96 callbackGasCostJuels) {
     _whenNotPaused();
 
-    Commitment memory commitment = s_requestCommitments[requestId];
+    if (keccak256(abi.encode(commitment)) != s_requestCommitments[commitment.requestId]) {
+      resultCode = 2; // FulfillResult.INVALID_REQUEST_ID
```

Comment:
I don't think that's a valid code here, can we introduce a new one?

Reply:
Yeah I think that makes more sense. Changing

Reply:
I think we can actually not even have requestId as a separate piece of the report now